assets/src/data/services/apollo/withApollo.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { ApolloProvider } from '@apollo/react-hooks';
+
+import { getClient } from './client';
+
+const withApollo = <P extends {}>(Component: React.ComponentType<P>): React.FC<P> => {
+	const WrappedComponent: React.FC<P> = (props) => {
+		return (
+			<ApolloProvider client={getClient()}>
+				<Component {...props} />
+			</ApolloProvider>
+		);
+	};
+
+	return WrappedComponent;
+};
+
+export default withApollo;
```

Comment:
will have to think at some point in the future how to make a check such that this provider isn't wrapping multiple time the same component, for example parent can be wrapped and then also some of the nested child might also be wrapped

Reply:
I'm inclined to try and do something like the following:

- create a single entry point for the application

- detect the current domain / route / use case

- load the appropriate base component for that request

- wrap that base component in data layer provider as well as any others (error, handling,etc) as required for that specific domain / route / use case